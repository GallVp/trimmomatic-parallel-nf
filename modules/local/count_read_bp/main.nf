process COUNT_READ_BP {
    tag "$meta.id"
    label 'process_single'

    conda "${moduleDir}/environment.yml"
    container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
        'https://depot.galaxyproject.org/singularity/ubuntu:22.04' :
        'nf-core/ubuntu:22.04' }"

    
    input:
    tuple val(meta), path(reads)

    output:
    tuple val(meta), path("*.log")  , emit: log
    path "versions.yml"             , emit: versions
    
    when:
    task.ext.when == null || task.ext.when
    
    script:
    def prefix = task.ext.prefix ?: "$meta.id"
    """
    touch ${prefix}.log

    for i in $reads; do
        gzip -dc \$i \\
            | awk \\
            'NR%4==2 { c++; l+=length(\$0) } END \\
            { print "File name: '\$i'"; print "Number of reads: "c; print "Number of bases in reads: "l }' \\
            &>> ${prefix}.log
    done

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        awk: "\$(awk -W version |& head -n1)"
    END_VERSIONS
    """
}