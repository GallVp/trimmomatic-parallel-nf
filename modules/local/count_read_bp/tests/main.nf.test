nextflow_process {

    name "Test Process COUNT_READ_BP"
    script "../main.nf"
    process "COUNT_READ_BP"
    
    tag "modules"
    tag "count_read_bp"

    test("Paired-Reads") {
        when {
            process {
                """
                input[0] = [
                    [ id: 'test', single_end:false ],
                    [
                        file(params.modules_testdata_base_path + 'genomics/sarscov2/illumina/fastq/test_1.fastq.gz', checkIfExists: true),
                        file(params.modules_testdata_base_path + 'genomics/sarscov2/illumina/fastq/test_2.fastq.gz', checkIfExists: true)
                    ]
                ]
                """
            }
        }

        then {
            assertAll (
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }
    }

}