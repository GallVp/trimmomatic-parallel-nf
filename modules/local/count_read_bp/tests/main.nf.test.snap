{
    "Paired-Reads": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test.log:md5,521cdba0e3b200ddce6c75ae99dc05b2"
                    ]
                ],
                "1": [
                    "versions.yml:md5,7b22f0c1a26dc96b2b2154bb1ac80fc5"
                ],
                "log": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test.log:md5,521cdba0e3b200ddce6c75ae99dc05b2"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,7b22f0c1a26dc96b2b2154bb1ac80fc5"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.0",
            "nextflow": "24.10.3"
        },
        "timestamp": "2025-01-23T21:34:52.773192"
    }
}